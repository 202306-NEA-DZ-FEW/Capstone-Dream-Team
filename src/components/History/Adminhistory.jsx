import React from "react";
import Historybar from "./Historybar";
import Historycard from "./Historycard";

export default function Adminhistory() {
    return (
        <>
            {/* <Historybar /> */}
            <Historycard />
        </>

        // <div className='flex-col'>
        //     <Historybar />
        //     <div className='flex-col'>
        //         <div className='mb-4'>
        //             <Historycard />
        //         </div>
        //     </div>
        // </div>
    );
}
