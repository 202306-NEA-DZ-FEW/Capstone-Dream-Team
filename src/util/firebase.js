// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
