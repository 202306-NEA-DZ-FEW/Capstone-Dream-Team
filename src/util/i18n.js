import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
