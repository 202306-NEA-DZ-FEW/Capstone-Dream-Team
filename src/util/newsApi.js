export async function fetchData(apiRoute) {
    const url = `https://newsapi.org/v2/${apiRoute}&apiKey=${process.env.NEXT_PUBLIC_NEWS_API}`;
    const response = await fetch(url);
    const data = await response.json();
    return data;
}
