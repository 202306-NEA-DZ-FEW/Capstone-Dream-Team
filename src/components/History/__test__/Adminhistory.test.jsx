import { I18nextProvider } from "react-i18next";

import renderer from "react-test-renderer";
import i18n from "../../../util/i18n";

import Adminhistory from "../Adminhistory";

import { getAuth } from "@/util/firebase";

// Mock the Firebase services
jest.mock("@/util/firebase", () => ({
    getAuth: jest.fn(),
    initializeApp: jest.fn(), // Mock initializeApp to avoid Firebase initialization
}));

// Mock the useRouter hook
jest.mock("next/router", () => ({
    useRouter: jest.fn().mockReturnValue({
        push: jest.fn(),
    }),
}));

it("renders correctly", () => {
    getAuth.mockReturnValue();

    const tree = renderer
        .create(
            <I18nextProvider i18n={i18n}>
                <Adminhistory language='en' />
            </I18nextProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
