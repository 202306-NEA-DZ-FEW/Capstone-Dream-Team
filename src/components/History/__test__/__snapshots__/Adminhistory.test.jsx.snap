// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="overflow-x-auto"
>
  <div
    className="inline-block min-w-full py-2 "
  >
    <div
      className="overflow-hidden"
    >
      <table
        className="min-w-full text-center text-sm font-light"
      >
        <thead
          className="border-b bg-neutral-800 font-medium text-white dark:border-neutral-500 dark:bg-neutral-900"
        >
          <tr>
            <th
              className=" px-6 py-4"
              scope="col"
            >
              history.Doner
            </th>
            <th
              className=" px-6 py-4"
              scope="col"
            >
              history.Meal
            </th>
            <th
              className=" px-6 py-4"
              scope="col"
            >
              history.Quantity
            </th>
            <th
              className=" px-6 py-4"
              scope="col"
            >
              history.Price
            </th>
            <th
              className=" px-6 py-4"
              scope="col"
            >
              history.Date
            </th>
            <th
              className=" px-6 py-4"
              scope="col"
            >
              history.UsedMeals
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
